Updating REST Transport Documentation (#12686)
CDCgov/prime-reportstream: prime-router/docs/onboarding-users/transport/rest.md

    - authType:       Authentication type i.e two-legged.  It is default to null (API shared key)
    - tlsKeystore:    The name for the credential manager to get the JKS used in TLS/SSL
    - parameters:     The map of parameters to be sent in the message (use with two-legged OAuth.  See ETOR RESTTransport)
    - authHeaders:    The key:value of authentication header
        . Authentication-Type:  Can be as following
            "Basic Auth" - Authenticates with string "Basic + base64(username+password)" to authentication header
            "username/password" - Authenticates with json body { "username": "<username>", "password": "<password>" }

>             "email/password" - Authenticates with json body { "email": "<username>", "password": "<password>" }

Reviewer: ```suggestion
            "email/password" - Authenticates with json body { "email": "<email>", "password": "<password>" }
```

Author: Thank you @adegolier

        . Content-Type: Can be as following
            "application/json" - Authentication body as json
            "text/plain" - Authentication body as text/plain

>             "Subscription" - Given to us by STLT (It is as plublic key and not secrete)

Reviewer: ```suggestion
            "Subscription" - Given to us by STLT (It is as public key and not secret)
```

        . ExpectSuccess: Set to true for expecting success and if not it will throw exception
        . Host: Set to host domain name.
    - headers:        The map of headers to be sent in the message
        . Host: Set to host domain name.
        . Content-Type: Can be as following
            "application/json" - Content/message body as json
            "text/plain" - Content/message body as text/plain
            "multipart/form-data"  - Content/message body as multipart/form-data or file

>             "Subscription" - Given to us by STLT (It is as plublic key and not secrete)

Reviewer: ```suggestion
            "Subscription" - Given to us by STLT (It is as public key and not secret)
```

        . Key: If Content-Type is "multipart/form-data", it Can be as following

>             "payload" - Set "multipart/form-data" key value to payload (uses by NY-PHD)

Reviewer: ```suggestion
            "payload" - Set "multipart/form-data" key value to payload (used by NY-PHD)
```

>             "files" - Set "multipart/form-data" key value to file (uses by LA-PHL/Natus).

Reviewer: ```suggestion
            "files" - Set "multipart/form-data" key value to file (used by LA-PHL/Natus).
```

        . Content-Type: Can be as following
            "application/json" - Authentication body as json
            "text/plain" - Authentication body as text/plain
            "Subscription" - Given to us by STLT (It is as public key and not secrete)
        . ExpectSuccess: Set to true for expecting success and if not it will throw exception
        . Host: Set to host domain name.
    - headers:        The map of headers to be sent in the message
        . Host: Set to host domain name.
        . Content-Type: Can be as following
            "application/json" - Content/message body as json
            "text/plain" - Content/message body as text/plain
            "multipart/form-data"  - Content/message body as multipart/form-data or file
            "Subscription" - Given to us by STLT (It is as public key and not secrete)
        . Key: If Content-Type is "multipart/form-data", it Can be as following
            "payload" - Set "multipart/form-data" key value to payload (used by NY-PHD)
            "files" - Set "multipart/form-data" key value to file (used by LA-PHL/Natus).

>         . File-Name: Set customer message's file name.  If the name includes substring "withdate", it will include datetime with the name.

Reviewer 2: Is this still needed?
